File: solutions/csharp/grains/1/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (!(n is >= 1 and <= 64))
            throw new ArgumentOutOfRangeException();

        return 1ul << (n - 1);
    }

    public static ulong Total()
        => UInt64.MaxValue;
}

File: solutions/c/grains/1/grains.c
#include "grains.h"


#define NUM_SQUARES    64

uint64_t square(uint8_t index) {
    if (index < 1 || index > NUM_SQUARES) return 0;
    return 1ULL << (index - 1);
}

uint64_t total(void) {
    return (uint64_t)-1;
}